modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/TopicUserAuthorizationTestCase.java

```diff
@@ -0,0 +1,584 @@
+     */
+    @AfterClass()
+    public void cleanUpTopics()
+            throws XPathExpressionException, LoginAuthenticationExceptionException, IOException,
+                   XMLStreamException, URISyntaxException, SAXException,
+                   TopicManagerAdminServiceEventAdminExceptionException,
+                   LogoutAuthenticationExceptionException {
+        LoginLogoutClient loginLogoutClientForUser = new LoginLogoutClient(this.automationContext);
+        String sessionCookie = loginLogoutClientForUser.login();
+        TopicAdminClient topicAdminClient =
+                new TopicAdminClient(this.backendURL, sessionCookie, ConfigurationContextProvider
+                        .getInstance().getConfigurationContext());
+        topicAdminClient.removeTopic("authTopic1");
+        topicAdminClient.removeTopic("authTopic2");
+        topicAdminClient.removeTopic("authTopic3");
+        topicAdminClient.removeTopic("authTopic4");
+        topicAdminClient.removeTopic("authTopic5");
+        topicAdminClient.removeTopic("authTopic6");
+        topicAdminClient.removeTopic("authTopic7");
+        topicAdminClient.removeTopic("authTopic8");
+        topicAdminClient.removeTopic("authTopic9");
+
+        loginLogoutClientForUser.logout();
+
+    }
+
+    /**
+     * Runs a test case where a consumer and publisher is created and published with a given user
+     * key from the automation.xml.
+     *
+     * @param userKey         The user key mentioned in the automation.xml for a specific user.
+     * @param destinationName The destination name of the topic.
+     * @throws XPathExpressionException
+     * @throws org.wso2.mb.integration.common.clients.exceptions.AndesClientConfigurationException
+     * @throws IOException
+     * @throws javax.jms.JMSException
+     * @throws org.wso2.mb.integration.common.clients.exceptions.AndesClientException
+     * @throws javax.naming.NamingException
+     */
+    private void createPublishAndSubscribeFromUser(String userKey, String destinationName)
+            throws XPathExpressionException, AndesClientConfigurationException, IOException,
+                   JMSException, AndesClientException, NamingException {
+        long sendCount = 10L;
+        long expectedCount = 10L;
+
+        AutomationContext userAutomationContext =
+                new AutomationContext("MB", "mb001", FrameworkConstants.SUPER_TENANT_KEY, userKey);
+        User contextUser = userAutomationContext.getContextTenant().getContextUser();
+
+        // Creating a consumer client configuration
+        AndesJMSConsumerClientConfiguration
+                consumerConfig =
+                new AndesJMSConsumerClientConfiguration(contextUser
+                                                                .getUserNameWithoutDomain(), contextUser
+                                                                .getPassword(), "127.0.0.1", 5672, ExchangeType.TOPIC, destinationName);
```

Comment:
Added a new constructor so that the configuration will automatically use the constants of host and port.
Fixed in https://github.com/hemikak/product-mb/commit/0c5b0a8a444a2c8dabf347f40e34698c0a980017